### ISSUE TITLE
[question] How to enable compression for all response?

### ISSUE THREAD

**Issue by lkwg82** [h2o/h2o#1269]
I do know, that I can enable compression via 
- header `x-compress-hint`
- specific mimetypes

But how do I enable for all mimetypes, even unknown ones?

Cheer Lars

**Comment by kazuho** [comment]
Setting `x-compress-hint: on` will enforce H2O to compress all the responses.

It is when the header does not exist (or when the value is `auto`) that the mime-type is taken into consideration.

**Comment by lkwg82** [comment]
Yes, this I know. This needs to change all backend servers to send this header.

How do I do this, when my `h2o` acts as reverse proxy and I want to set it at one place?

**Comment by deweerdt** [comment]
You will need a new configuration knob to do this, AFAIK. One way to do it would be to add an `ALWAYS` option to `on_config_compress`, change `h2o_compress_args_t::compress_defaults_to_off` to a tri-state (`on`, `off`, `always`), and break from the switch in `lib/handler/compress.c::on_setup_ostream` if the know is set to `always`.

**Comment by lkwg82** [comment]
@kazuho Is there any way to make such code change being merged, or is this conceptually not wanted?

**Comment by lkwg82** [comment]
@dewerdt thx for your hint

**Comment by lkwg82** [comment]
@deweerdt thx for your hint
